test/gtest/uct/test_peer_failure.cc

```diff
@@ -263,21 +277,25 @@ UCS_TEST_SKIP_COND_P(test_uct_peer_failure, purge_failed_peer,
     send_recv_am(0);
     send_recv_am(1);
 
-    const size_t num_pend_sends = 3ul;
-    uct_pending_req_t reqs[num_pend_sends];
+    const ucs_time_t loop_end_limit = ucs::get_deadline();
+    const size_t num_pend_sends     = 64ul;
+    pending_send_request_t *reqs    = new pending_send_request_t[num_pend_sends];
+    const uint64_t send_data        = 0;
     {
         scoped_log_handler slh(wrap_errors_logger);
 
-        kill_receiver();
-
         ucs_status_t status;
         do {
-            status = uct_ep_am_short(ep0(), 0, 0, NULL, 0);
-        } while (status == UCS_OK);
+            status = uct_ep_am_short(ep0(), 0, 0, &send_data,
+                                     sizeof(send_data));
+        } while ((status == UCS_OK) && (ucs_get_time() < loop_end_limit));
+
```

Comment:
maybe skip the test here if all  am_shorts completed OK?
Otherwise you will not be able to add to pending

Reply:
done